apps/server/src/shared/testing/factory/domainobject/tool/external-tool-datasheet-template-data.factory.ts

```diff
@@ -0,0 +1,76 @@
+import {
+	CustomParameterLocation,
+	LtiMessageType,
+	LtiPrivacyPermission,
+	ToolConfigType,
+} from '@modules/tool/common/enum';
+import { DeepPartial } from 'fishery';
+import {
+	ExternalToolDatasheetTemplateData,
+	ExternalToolParameterDatasheetTemplateData,
+} from '@modules/tool/external-tool/domain';
+import { DoBaseFactory } from '../do-base.factory';
+
+class ExternalToolParameterDatasheetTemplateDataFactory extends DoBaseFactory<
+	ExternalToolParameterDatasheetTemplateData,
+	ExternalToolParameterDatasheetTemplateData
+> {}
+
+export const externalToolParameterDatasheetTemplateDataFactory =
+	ExternalToolParameterDatasheetTemplateDataFactory.define(
+		ExternalToolParameterDatasheetTemplateData,
+		({ sequence }) => {
+			return {
+				name: `custom-parameter-${sequence}`,
+				properties: '',
+				type: 'Zeichenkette',
+				scope: 'Schule',
+				location: CustomParameterLocation.BODY,
+			};
+		}
+	);
+
+export class ExternalToolDatasheetTemplateDataFactory extends DoBaseFactory<
```

Comment:
your ExternalToolDatasheetTemplateData is not a (legacy) DO, so it should not use the DoBaseFactory. Use the fishery Factory please.